processor/src/networks/bitcoin.rs

```diff
@@ -450,6 +477,85 @@ impl Bitcoin {
       }
     }
   }
+
+  // expected script has to start with SHA256 PUSH MSG_HASH OP_EQ ..
+  fn expected_script_pattern(&self, script: &ScriptBuf) -> Option<bool> {
+    let mut ins = script.instructions();
+
+    // first item should be SHA256 code
+    if ins.next()?.ok()?.opcode()? != OP_SHA256 {
+      return Some(false);
+    }
+
+    // next should be a data push
+    ins.next()?.ok()?.push_bytes()?;
+
+    // next should be a equality check
+    if ins.next()?.ok()?.opcode()? != OP_EQUAL {
+      return Some(false);
+    }
+
+    Some(true)
+  }
+
+  async fn spent_output_of(&self, input: &TxIn) -> TxOut {
```

Comment:
`_of` doesn't necessarily make grammatical sense here. This returns the spent output and that alone make sense.